pkg/apis/pingcap/v1alpha1/tidb_config_wraper.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	stdjson "encoding/json"
+	"reflect"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb-operator/pkg/util/config"
+	"github.com/pingcap/tidb-operator/pkg/util/toml"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+var _ stdjson.Marshaler = &TiDBConfigWraper{}
+var _ stdjson.Unmarshaler = &TiDBConfigWraper{}
+
+func NewTiDBConfig() *TiDBConfigWraper {
+	return &TiDBConfigWraper{
+		GenericConfig: config.New(map[string]interface{}{}),
+	}
+}
+
+type TiDBConfigWraper struct {
+	Deprecated *TiDBConfig
```

Comment:
Do we need the field `Deprecated`? I think we can use a temporary variable at https://github.com/pingcap/tidb-operator/pull/3327/files#diff-0ee0fff3f29e60ffc99a7001b4a6f781R70

Reply:
Deprecated